src/Groups/gsets.jl

```diff
@@ -25,16 +30,307 @@ export
+          push!(orb, img)
+        end
+      end
+    end
+
+    res = as_gset(Omega.group, Omega.action_function, orbarray)
+    # We know that this G-set is transitive.
+    AbstractAlgebra.set_special(res, :orbits => [orbarray])
+    return res
+end
+
+
+#TODO: - Currently this conflicts with a method in
+#        `experimental/GaloisGrp/Group.jl`
+#      - Provide more convenience methods for "natural" actions",
+#        as for `gset`?
+# Hecke.orbit(G::PermGroup, pt::T) where {T<:Union{Base.Integer,fmpz}} = GSetByElements(G, [pt])
+
+
+#############################################################################
+##
+##  `:orbits` an array of G-sets
+
+function orbits(Omega::GSetByElements{<:GAPGroup})
+    orbs = AbstractAlgebra.get_special(Omega, :orbits)
+    if orbs == nothing
+      G = Omega.group
+      orbs = []
+      for p in Omega.seeds
+        if all(o -> !(p in o), orbs)
+          push!(orbs, Hecke.orbit(Omega, p))
+        end
+      end
+      AbstractAlgebra.set_special(Omega, :orbits => orbs)
+    end
+    return orbs
+end
+
+
+#############################################################################
+##
+##  `:elements` an array of points
+
+function elements(Omega::GSetByElements)
+    elms = AbstractAlgebra.get_special(Omega, :elements)
+    if elms == nothing
+      orbs = orbits(Omega)
+      elms = union(map(elements, orbs)...)
+      AbstractAlgebra.set_special(Omega, :elements => elms)
+    end
+    return elms
+end
+
+
+#############################################################################
+##
+
+# In fact, '<:GAPGroup' is not used at all in this function.
+function permutation(Omega::GSetByElements{T}, g::BasicGAPGroupElem{T}) where T<:GAPGroup
+    omega_list = GAP.julia_to_gap(elements(Omega))
+    gfun = GAP.julia_to_gap(Omega.action_function)
+
+    # The following works only because GAP does not check
+    # whether the given group element 'g' is a group element.
+    pi = GAP.Globals.PermutationOp(g, omega_list, gfun)
+
+    sym = AbstractAlgebra.get_special(Omega, :action_range)
+    if sym == nothing
+      sym = symmetric_group(length(Omega))
+      AbstractAlgebra.set_special(Omega, :action_range => sym )
+    end
+    return group_element(sym, pi)
+end
+
+
+############################################################################
+##
+##  action homomorphisms
+
+# The following must be executed at runtime.
+# (Eventually put it into some `__init__`.)
+function __init_JuliaData()
+    if ! GAP.Globals.ISB_GVAR(GAP.julia_to_gap("JuliaData"))
+      GAP.evalstr("DeclareAttribute( \"JuliaData\", IsObject );");
+      GAP.evalstr("InstallOtherMethod( ImagesRepresentative, " *
+      "[ IsActionHomomorphism and HasJuliaData, IsMultiplicativeElementWithInverse ], " *
+      "function( hom, elm ) " *
+      "local data; " *
+      "data:= JuliaData( hom ); " *
+      "return Julia.Oscar.permutation(data[1], Julia.Oscar.group_element(data[2], elm)).X;" *
+      "end );")
```

Comment:
Perhaps use a triple quoted string? Those can contain newlines, so one can copy&paste between here and a terminal.

Reply:
Yes, this would be a quick improvement.
In the long run, the question is where to put such "manipulations of GAP":
If `JuliaData` turns out to be useful in general, `JuliaInterface` is a natural place for its declaration.
The method used here depends on Oscar, thus `JuliaInterface` is not an option.
(Ideally, standard GAP could be changed/improved such that Oscar can just use it.)